lib/plugins/usermanager/cli.php

```diff
@@ -0,0 +1,279 @@
+            return 1;
+        }
+
+        return 0;
+    }
+
+    /**
+     * Deletes users
+     * @param array $args
+     * @return int
+     */
+    protected function cmdDelete(array $args)
+    {
+        global $auth;
+
+        if (!$auth->canDo('delUser')) {
+            $this->error($this->getLang('nosupport'));
+            return 1;
+        }
+
+        $users = explode(',', $args[0]);
+        $count = $auth->triggerUserMod('delete', array($users));
+
+        if (!($count == count($users))) {
+            $part1 = str_replace('%d', $count, $this->getLang('delete_ok'));
+            $part2 = str_replace('%d', (count($users) - $count), $this->getLang('delete_fail'));
+            $this->error("$part1, $part2");
+
+            return 1;
+        }
+
+        return 0;
+    }
+
+    /**
+     * Adds an user to group(s)
+     *
+     * @param array $args
+     * @return int
+     */
+    protected function cmdAddToGroup(array $args)
+    {
+        global $auth;
+
+        list($name, $newgrps) = $args;
+        $newgrps = array_filter(array_map('trim', explode(',', $newgrps)));
+        $oldinfo = $auth->getUserData($name);
+        $changes = array();
+
+        if (!empty($newgrps) && $auth->canDo('modGroups')) {
+            $changes['grps'] = $oldinfo['grps'];
+            foreach ($newgrps as $group) {
+                if (!in_array($group, $oldinfo['grps'])) {
+                    array_push($changes['grps'], $group);
+                }
+            }
+        }
+
+        if (!empty(array_diff($changes['grps'], $oldinfo['grps']))) {
+            if ($auth->triggerUserMod('modify', array($name, $changes))) {
+                $this->info($this->getLang('update_ok'));
+            } else {
+                $this->error($this->getLang('update_fail'));
+                return 1;
+            }
+        }
+
+        return 0;
+    }
+
+    /**
+     * Removes an user from group(s)
+     *
+     * @param array $args
+     * @return int
+     */
+    protected function cmdRemoveFromGroup(array $args)
+    {
+        global $auth;
+
+        list($name, $grps) = $args;
+        $grps = array_filter(array_map('trim', explode(',', $grps)));
+        $oldinfo = $auth->getUserData($name);
+        $changes = array();
+
+        if (!empty($grps) && $auth->canDo('modGroups')) {
+            $changes['grps'] = $oldinfo['grps'];
+            foreach ($grps as $group) {
+                if (($pos = array_search($group, $changes['grps'])) == !false) {
+                    unset($changes['grps'][$pos]);
+                }
+            }
+        }
+
+        if (!empty(array_diff($oldinfo['grps'], $changes['grps']))) {
+            if ($auth->triggerUserMod('modify', array($name, $changes))) {
+                $this->info($this->getLang('update_ok'));
+            } else {
+                $this->error($this->getLang('update_fail'));
+                return 1;
+            }
+        }
+
+        return 0;
+    }
```

Comment:
I added a user to `foobar`, then to `foobar2`. Then removing him from `foobar` doesn't work. I'm not sure why. It neither gives an error nor a success message.